homeassistant/components/asuswrt/sensor.py

```diff
@@ -96,12 +123,10 @@ async def async_setup_entry(
         coordinator = sensor_data[KEY_COORDINATOR]
         sensors = sensor_data[KEY_SENSORS]
         for sensor_key in sensors:
-            if sensor_key in CONNECTION_SENSORS:
-                entities.append(
-                    AsusWrtSensor(
-                        coordinator, router, sensor_key, CONNECTION_SENSORS[sensor_key]
-                    )
-                )
+            for sensor_descr in CONNECTION_SENSORS:
+                if sensor_descr.key == sensor_key:
+                    entities.append(AsusWrtSensor(coordinator, router, sensor_descr))
+                    break
```

Comment:
Use the following code to make the complexity N instead of N*N

```python
async_add_entities([
    AsusWrtSensor(coordinator, router, sensor_descr)
    for sensor_descr in CONNECTION_SENSORS
    if sensor_descr.key in sensors
], True)
```

Reply:
Fixed